src/debug.cpp

```diff
@@ -777,4 +781,90 @@ std::ostream &DebugLog( DebugLevel lev, DebugClass cl )
     return nullStream;
 }
 
+std::string game_info::operating_system()
+{
+#if defined(__ANDROID__)
+    return "Android";
+#elif defined(_WIN32)
+    return "Windows";
+#elif defined(__linux__)
+    return "Linux";
+#elif defined(unix) || defined(__unix__) || defined(__unix) || defined(__APPLE__) && defined(__MACH__) // unix; BSD; MacOs
+#if defined(__APPLE__) && defined(__MACH__)
+#include <TargetConditionals.h>
```

Comment:
Putting a `#include` in the middle of a function scares me somewhat.  If you're confident this is safe could you add a comment to explain why.

Reply:
First of all thank you very much for your review!

I let the include here and put a comment on the "why". It says:

```
    // The following include is **only** needed for the TARGET_xxx defines below and is only included if both of the above defines are true. 
    // The whole function only relying on compiler defines, it is probably more meaningful to include it here and not mingle with the
    // headers at the top of the .cpp file.
```

I'm also not really happy with includes in a middle of a file, but in this case the whole function relies on compiler defines; I think this might be one of the few exception where it makes sense to have the include in the function. 

If it really poses a problem, I can move it to the top.